packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/TaskService.java

```diff
@@ -431,15 +443,33 @@ private void onTaskResponseOneInstanceReceives(final Message message) {
 			try {
 				// create the notification event
 				final NotificationEvent event = new NotificationEvent();
-				event.setData(resp);
+				event.setData(TaskNotificationEventData.createFrom(resp));
 
 				log.info("Creating notification event under group id: {}", resp.getId());
 
 				notificationService.createNotificationEvent(resp.getId(), event);
+
 			} catch (final Exception e) {
 				log.error("Failed to persist notification event for for task {}", resp.getId(), e);
 			}
 
+			try {
```

Comment:
This code needs to be moved from `onTaskResponseOneInstanceReceives` method  to the `onTaskResponseAllInstanceReceive` method.